lock/README.md

```diff
@@ -0,0 +1,57 @@
+# lock
+
+## Prerequisites
+
+* [winglang](https://winglang.io).
+
+## Installation
+
+`sh
+npm i @winglibs/lock
+`
+
+## Usage
+
+```js
+bring lock;
+bring cloud;
+bring util;
+
+let l = new lock.Lock();
+let q = new cloud.Queue();
+
+q.setConsumer(inflight (m: str) => {
+  let id = util.uuidv4();
+  l.acquire("my-lock", 20s);
+  log("{id}:start");
+  util.sleep(1s);
+  log("{id}:end");
+  l.release("my-lock");
```

Comment:
Does this mean anybody can release anyone else's lock while it is still being held? Is there a way we can prevent this?

Reply:
Interesting 🤔 

Do you think this is a better API? 
```
let myLock = l.lock("name");
myLock.unlock();
```

Reply:
I prefer the Java-style API, but it could be a familiarity thing. In my head, if a thread or process or inflight isn't obtaining an exclusive handle when it acquires a lock, then it's less of a "lock" and more of an atomic flag, or mutex of some sort.

I'm not all that sure though so I'm ok trying this out